src/effects/huff_puff_breath.c

```diff
@@ -8,7 +8,46 @@ INCLUDE_ASM(s32, "effects/huff_puff_breath", huff_puff_breath_main);
 void huff_puff_breath_init(void) {
 }
 
-INCLUDE_ASM(s32, "effects/huff_puff_breath", huff_puff_breath_update);
+void huff_puff_breath_update(EffectInstance* effect) {
+    HuffPuffBreathFXData* data = effect->data.huffPuffBreath;
+    s32 temp_a1;
+    s32 temp_a2;
+
+    if (effect->flags & 16) {
+        effect->flags &= ~16;
+        data->unk_10 = 16;
+    }
+    if (data->unk_10 < 1000) {
+        data->unk_10--;
+    }
+    temp_a1 = data->unk_10;     // Needed for match?
+    temp_a2 = data->unk_14 + 1; // Needed for match?
+    data->unk_14++;
+    if (temp_a1 < 0) {
+        shim_remove_effect(effect);
+        return;
+    }
+    if (temp_a1 < 16) {
+        data->unk_24 = temp_a1 * 16;
+    }
+    if (temp_a2 < 16) {
+        data->unk_24 = (temp_a2 * 16) + 15;
+    }
```

Comment:
```suggestion
    temp_a2 = ++data->unk_14;
    
    if (data->unk_10 < 0) {
        shim_remove_effect(effect);
        return;
    }
    if (data->unk_10 < 16) {
        data->unk_24 = data->unk_10 * 16;
    }
    if (temp_a2 < 16) {
        data->unk_24 = (temp_a2 * 16) + 15;
    }
```

[See here](https://decomp.me/scratch/hezDG)

Reply:
I'd prefer postincrementing the field and then storing to a temp. I'm gonna play with this locally and see if I can clean it up (also applies above)

Reply:
additionally, unk14 is lifeTime and unk_10 is timeLeft

Reply:
Oh, and you can remove the declaration for `temp_a1` on line 13